sql/randgen: limit the size of random geometry / geography further

fqazi opened the pull request:
Previously, the random schema changer workload could fail if gigantic datums were generated by randgen for geometry / geography types. We did add a size limit to constrain the size of these types, but it was still to excessive. To take it a step further we are going to limit the size of these types to 4 megabytes.

Fixes: #139494

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139702)
<!-- Reviewable:end -->

fqazi committed:
workload/schemachanger: limit size of datums during insert

Previously, the insert statement in the randomized schema changer
workload could have unbounded sizes for individual datums. This patch
adds size limit for datums in the insert operation, where datums will be
regenerated a limited number of times until they are below a maximum
size.

Fixes: #139494

Release note: None
